native/core/src/models/status/membership.rs

```diff
@@ -1,147 +1,89 @@
+            Some(SdkMembershipChange::InvitationRejected) => "invitationRejected",
+            Some(SdkMembershipChange::InvitationRevoked) => "invitationRevoked",
+            Some(SdkMembershipChange::Knocked) => "knocked",
+            Some(SdkMembershipChange::KnockAccepted) => "knockAccepted",
+            Some(SdkMembershipChange::KnockRetracted) => "knockRetracted",
+            Some(SdkMembershipChange::KnockDenied) => "knockDenied",
+            Some(SdkMembershipChange::None)
+            | Some(SdkMembershipChange::Error)
+            | Some(SdkMembershipChange::NotImplemented)
+            | None => {
+                return Err(());
+            }
+        };
+        Ok(MembershipContent {
+            user_id: value.user_id().to_owned(),
+            change: change.to_owned(),
+        })
+    }
 }
 
-impl TryFrom<RumaMembershipChange<'_>> for MembershipChangeType {
+impl TryFrom<(RumaMembershipChange<'_>, OwnedUserId)> for MembershipContent {
     type Error = ();
 
-    fn try_from(value: RumaMembershipChange<'_>) -> Result<Self, ()> {
-        Ok(match value {
-            RumaMembershipChange::Banned => MembershipChangeType::Banned,
-            RumaMembershipChange::Joined => MembershipChangeType::Joined,
-            RumaMembershipChange::Left => MembershipChangeType::Left,
-            RumaMembershipChange::Unbanned => MembershipChangeType::Unbanned,
-            RumaMembershipChange::Kicked => MembershipChangeType::Kicked,
-            RumaMembershipChange::Invited => MembershipChangeType::Invited,
-            RumaMembershipChange::KickedAndBanned => MembershipChangeType::KickedAndBanned,
-            RumaMembershipChange::InvitationAccepted => MembershipChangeType::InvitationAccepted,
-            RumaMembershipChange::InvitationRejected => MembershipChangeType::InvitationRejected,
-            RumaMembershipChange::InvitationRevoked => MembershipChangeType::InvitationRevoked,
-            RumaMembershipChange::Knocked => MembershipChangeType::Knocked,
-            RumaMembershipChange::KnockAccepted => MembershipChangeType::KnockAccepted,
-            RumaMembershipChange::KnockRetracted => MembershipChangeType::KnockRetracted,
-            RumaMembershipChange::KnockDenied => MembershipChangeType::KnockDenied,
-            RumaMembershipChange::ProfileChanged {
-                displayname_change,
-                avatar_url_change,
-            } => MembershipChangeType::ProfileChanged {
-                displayname_change: displayname_change.map(|c| Change {
-                    old: c.old.map(ToOwned::to_owned),
-                    new: c.new.map(ToOwned::to_owned),
-                }),
-                avatar_url_change: avatar_url_change.map(|c| Change {
-                    old: c.old.map(ToOwned::to_owned),
-                    new: c.new.map(ToOwned::to_owned),
-                }),
-            },
+    fn try_from(value: (RumaMembershipChange<'_>, OwnedUserId)) -> Result<Self, Self::Error> {
```

Comment:
same as above. let's keep the type until the very end so we can later expose that via uniffi

Reply:
This reason is the same as https://github.com/acterglobal/a3/pull/2790#discussion_r2037951350.